include/sys/spa.h

```diff
@@ -1183,6 +1183,11 @@ extern int spa_wait_tag(const char *name, zpool_wait_activity_t activity,
 extern void spa_notify_waiters(spa_t *spa);
 extern void spa_wake_waiters(spa_t *spa);
 
+extern void spa_create_os(spa_t *spa);
```

Comment:
This is a bit pedantic, but looking at where these are called (during create and import) perhaps it would make more sense to call this `spa_import_os()`?  This would also be symmetric with `spa_export_os` which is kind of nice.

Aside from that, this looks good and it's another bit of groundwork we can get in place.

Reply:
Yeah, import does fit better. It was due to the "whenever we create a spa*" but I think even creating a pool calls import on it.